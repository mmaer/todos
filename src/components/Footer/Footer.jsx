import React, { memo } from "react";

import Filters from "./Filters";

import { filterActiveTodo, filterCompletedTodo } from "../../utils";

import "./Footer.scss";

const Footer = memo(({ todos, removeCompletedTodos }) => {
  return (
    <footer className="footer">
      <span className="footer__todo-count">
        <strong>
          {todos.filter(filterActiveTodo).length}
          {` `}
        </strong>
        items left
      </span>
      <Filters />
      {!!todos.filter(filterCompletedTodo).length && (
        <button
          onClick={removeCompletedTodos}
          className="footer__clear-completed"
        >
          Clear completed
        </button>
      )}
    </footer>
  );
});

export default Footer;
