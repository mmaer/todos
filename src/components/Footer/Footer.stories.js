import React from "react";
import { MemoryRouter } from "react-router-dom";

import Footer from ".";

export default {
  title: "Footer",
};

const TODOS = [{ completed: false, active: true }];

export const footer = () => (
  <MemoryRouter>
    <Footer todos={TODOS} />
  </MemoryRouter>
);
