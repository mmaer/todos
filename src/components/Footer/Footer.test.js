import React from "react";
import { MemoryRouter } from "react-router-dom";
import { shallow, mount } from "enzyme";

import Footer from "./";

const TODOS = [{ completed: false }, { completed: false }, { completed: true }];

const mountWithRouter = Component => <MemoryRouter>{Component}</MemoryRouter>;

describe("Footer component", () => {
  it("Should match snapshot", () => {
    const wrapper = shallow(
      <Footer removeCompletedTodos={() => {}} todos={TODOS} />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it("Should not display 'Clear completed' button when there are no completed todos", () => {
    const mockRemoveCompletedTodosFunc = jest.fn();
    const wrapper = mount(
      mountWithRouter(
        <Footer
          removeCompletedTodos={mockRemoveCompletedTodosFunc}
          todos={[{ completed: false }]}
        />
      )
    );
    const button = wrapper.find(".footer__clear-completed");
    expect(button.length).toBe(0);
  });

  it("Should call removeCompletedTodos when 'Clear completed' is clicked", () => {
    const mockRemoveCompletedTodosFunc = jest.fn();
    const wrapper = mount(
      mountWithRouter(
        <Footer
          removeCompletedTodos={mockRemoveCompletedTodosFunc}
          todos={TODOS}
        />
      )
    );
    const button = wrapper.find(".footer__clear-completed");
    button.simulate("click");

    expect(mockRemoveCompletedTodosFunc.mock.calls.length).toBe(1);
  });
});
