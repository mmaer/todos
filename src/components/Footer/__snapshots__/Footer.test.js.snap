// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer component Should match snapshot 1`] = `
<footer
  className="footer"
>
  <span
    className="footer__todo-count"
  >
    <strong>
      2
       
    </strong>
    items left
  </span>
  <Memo() />
  <button
    className="footer__clear-completed"
    onClick={[Function]}
  >
    Clear completed
  </button>
</footer>
`;
